### PR TITLE
make syndicate elite hardsuits immune to bombs

### PR DESCRIPTION
**Pull request by Simplehorror**
# Document the changes in your pull request

crew members outnumber nukies
crewmembers can maxcap all nukies in a 1 for 5 trade to win

# Spriting

# Wiki Documentation

made syndicate elite hardsuits immune to bombs
# Changelog

:cl:  
tweak: made syndicate elite hardsuits immune to bombs
/:cl:

### CODE REVIEW

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -383,7 +383,7 @@
 	alt_desc = "An elite version of the syndicate helmet, with improved armour and fireproofing. It is in combat mode. Property of Gorlex Marauders."
 	icon_state = "hardsuit0-syndielite"
 	hardsuit_type = "syndielite"
-	armor = list(MELEE = 60, BULLET = 60, LASER = 50, ENERGY = 25, BOMB = 90, BIO = 100, RAD = 70, FIRE = 100, ACID = 100, WOUND = 25)
+	armor = list(MELEE = 60, BULLET = 60, LASER = 50, ENERGY = 25, BOMB = 1000, BIO = 100, RAD = 70, FIRE = 100, ACID = 100, WOUND = 25)
```
Comment (BurgerLUA) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Should be 1200 bomb resist, its more balanced that way.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -395,7 +395,7 @@
 	icon_state = "hardsuit0-syndielite"
 	hardsuit_type = "syndielite"
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/syndi/elite
-	armor = list(MELEE = 60, BULLET = 60, LASER = 50, ENERGY = 25, BOMB = 90, BIO = 100, RAD = 70, FIRE = 100, ACID = 100, WOUND = 25)
+	armor = list(MELEE = 60, BULLET = 60, LASER = 50, ENERGY = 25, BOMB = 1000, BIO = 100, RAD = 70, FIRE = 100, ACID = 100, WOUND = 25)
```
Comment (BurgerLUA) [inline code/modules/clothing/spacesuits/hardsuit.dm]: here too, maybe make this one have 35 wound resist too to also improve the balance

**Review by BurgerLUA — CHANGES_REQUESTED** [review]

**Comment by Moltijoe** [comment]
My god no, this isnt immunity, this turns bombs into an Aheal on them

**Comment by Simplehorror** [comment]
> My god no, this isnt immunity, this turns bombs into an Aheal on them

100 doesn't block devestation

**Comment by Moltijoe** [comment]
> 100 doesn't block devestation

100 bomb armour turns devastation from a gib into a total of 100 damage
going above 100 starts to break the damage formula
anything above 120 bomb armour will make them heal
in order for bomb armour to actually just be immunity, it needs to be exactly 120

**Comment by Simplehorror** [comment]
cant we just multiply it by 1.2 or some shit

**Comment by Moltijoe** [comment]
> cant we just multiply it by 1.2 or some shit

what? all i'm saying is that setting bomb armour to be 120 gives them immunity like you want, where as giving them 1000 like this pr currently has makes bombs heal them

**Comment by Sniblet** [comment]
Is there a compromise somewhere between preventing some scientist with a valve being the scariest living entity on the station, and allowing syndies to walk through their own grenades

**Comment by Moltijoe** [comment]
> Is there a compromise somewhere between preventing some scientist with a valve being the scariest living entity on the station, and allowing syndies to walk through their own grenades

not without redoing how explosion damage is calculated or adding a snowflake check

**Comment by chemistrymain2** [comment]
> > cant we just multiply it by 1.2 or some shit
> 
> what? all i'm saying is that setting bomb armour to be 120 gives them immunity like you want, where as giving them 1000 like this pr currently has makes bombs heal them

this is incorrect, it won't heal them

**Comment by Moltijoe** [comment]
> this is incorrect, it won't heal them

i suppose i've not tried it, but with how the damage calculates, you'd end up with a negative number
and at no point is there a check to make sure a negative number won't heal them

either way, 120 is perfect to make bombs deal exactly 0 damage without potentially causing other problems

**Comment by chemistrymain2** [comment]
> > this is incorrect, it won't heal them
> 
> i suppose i've not tried it, but with how the damage calculates, you'd end up with a negative number and at no point is there a check to make sure a negative number won't heal them
> 
> either way, 120 is perfect to make bombs deal exactly 0 damage without potentially causing other problems

with how damage is calculated, it'd just hit zero

you can't get a negative value ever

if you actually wanted to make bomb proof suits you'll have to redo explosion code on human defense, as they're hardcoded to do various effects even accounting for armor

**Comment by Moltijoe** [comment]
> with how damage is calculated, it'd just hit zero
> 
> you can't get a negative value ever

ah, yeah, looking at it i somehow missed this line in take_overall_damage
image


> if you actually wanted to make bomb proof suits you'll have to redo explosion code on human defense, as they're hardcoded to do various effects even accounting for armor
 
true, but it seemed like jamie wasn't keen on that seeing as he merged the pr that made bomb armour more effective rather than a proper percentage. currently the only bomb immune thing seems to be an ascended ash heretic, and even that might not be coded correctly to grant it.

**Comment by JamieD1** [comment]
lol

**Pull request closed without merging**